code/modules/mob/living/carbon/xenomorph/facehuggers.dm

```diff
@@ -232,6 +233,17 @@
 		return TRUE
 	return FALSE
 
+/obj/item/clothing/mask/facehugger/proc/check_barricade(datum/source, atom/target, range, thrower, spin)
+	var/mob/living/carbon/xenomorph/X = usr
+	var/list/turf_between = getline(X,target)
+	for(var/turf/T in turf_between)
+		var/obj/structure/barricade/found = locate() in T
+		if(found)
+			if(!found.density)
+				continue
+			to_chat(usr,"<span class='danger'>The facehugger isnt able to be thrown over the barricade!</span>")
```

Comment:
Please make this more 'lore firendly'

Perhaps "you're underhand throw slams the hugger into the [found.name]"

Reply:
Hugh it just drops at your feet, and I would have to edit throw_ats base atom proc to change that